### PR TITLE
Changes involving submit and change Handlers

### PR DESCRIPTION
**Pull request by KasliwalSwapnil**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ function App() {
       <Sidebar />
       <div className="container">
         <Routes>
-            <Route exact path="/" element={<FormElement obj = {config["config"]["apis"]["create"]} title = "Create Partner" description = "This Api helps you to create a partner in the system"/>}> 
+            <Route exact path="/" element={<FormElement obj = {config["config"]["apis"]["getSubscription"]} title = "Get Subscription ID" description = "This Api helps you to get the subscription id."/>}>
```
Comment (ayeshahotstar) [inline src/App.js]: /** TODO: 
* sub routing
**/

Comment (ayeshahotstar) [reply src/App.js]: error boundary to check config value https://reactjs.org/docs/error-boundaries.html

File: src/Components/FormElement.js [hunk 2]
```diff
@@ -1,38 +1,65 @@
...
+      method: 'GET',
+      headers,
+      mode: 'cors'
+         })
+         .then(results => console.log(results))}
+
+    catch(error){console.error(error)}
+
+  }
+  const [values, setValues] = useState({})
```
Comment (ayeshahotstar) [inline src/Components/FormElement.js:35]: const {obj:{pathParams, queryParams, headers}={}} = props
....
pathParams && (<></>)

Comment (ayeshahotstar) [reply src/Components/FormElement.js:35]: add prototypes for each component - https://reactjs.org/docs/typechecking-with-proptypes.html

File: src/Components/FormElement.js [hunk 3]
```diff
@@ -1,38 +1,65 @@
...
         <h2>{props.title}</h2>
         <br/>
         <h3>{props.description}</h3>
         <br/>
         <br/>
-        {props.obj["pathParams"] && props.obj["pathParams"].map((element, idx) => (
+        {props.obj["pathParams"] && props.obj["pathParams"].map((element) => (
           <>
-          <Input name = {element.name} inputType = {element.inputType} placeholder = {element.name} type={element.type} requestContentType={element.requestContentType}/>
+          <Input values = {values} setValues = {setValues} name = {element.name} inputType = {element.type} placeholder = {element.name} type={element.type} requestContentType={element.requestContentType}/>
```
Comment (ayeshahotstar) [inline src/Components/FormElement.js:46]: const mapInputValues = (element) => {
 const {name, type, requestContentType} = element
 return {
  name: name,
  placeholder: name
  .......
 }
}


....
const inputProps = mapInputProps(element)
<Input {...inputProps}/>

File: src/Components/FormElement.js [hunk 4]
```diff
@@ -1,38 +1,65 @@
...
+         .then(results => console.log(results))}
+
+    catch(error){console.error(error)}
+
+  }
+  const [values, setValues] = useState({})
   return (
     <form>
         <h2>{props.title}</h2>
         <br/>
```
Comment (ayeshahotstar) [inline src/Components/FormElement.js:39]: use css - padding and margins - https://www.w3schools.com/css/css_boxmodel.asp

File: src/Components/Input.js [hunk 5]
```diff
@@ -2,8 +2,25 @@ import React, {useState} from 'react'
 
 export default function Input(props) {
```
Comment (ayeshahotstar) [inline src/Components/Input.js:3]: props ={}

Comment (ayeshahotstar) [reply src/Components/Input.js:3]: //better
const {placeholder} = props || {}

File: src/Components/Input.js [hunk 6]
```diff
@@ -2,8 +2,25 @@ import React, {useState} from 'react'
...
     const [inputValue, setinputValue] = useState("");
+    const [value, setValue] = useState(JSON.stringify(props.placeholder));
+
     const handleOnChange = (event)=>{
-        setinputValue(event.target.value)
+        if(props.inputType === "JSON")
```
Comment (ayeshahotstar) [inline src/Components/Input.js:10]: handle default values

File: src/Components/FormElement.js (hunk 2 above)
Comment (KasliwalSwapnil) [reply src/Components/FormElement.js:35]: added proptypes and destructuring of the object is done.

File: src/Components/FormElement.js (hunk 3 above)
Comment (KasliwalSwapnil) [reply src/Components/FormElement.js:46]: Made the function but not used till now, because of the config changes in the main branch.

**Pull request merged**